crates/core/src/db/relational_db.rs

```diff
@@ -101,9 +102,10 @@ impl RelationalDB {
             if let Some(message_log) = &message_log {
                 let message_log = message_log.lock().unwrap();
                 let max_offset = message_log.open_segment_max_offset;
-                for message in message_log.iter() {
+                for commit in commit_log::Iter::from(message_log.segments()) {
+                    let commit = commit?;
+
                     segment_index += 1;
```

Comment:
@aasoni Unsure why this is named `segment_index` when it counts commits (i.e. messages). If it is _supposed_ to count commits, this could be further simplified via `.enumerate()`.

Reply:
this may be a question for @cloutiertyler, I've only rearranged things in this function and added debug logging. Not sure why the original naming decision

Reply:
Okay... I guess I'll just leave it like that, then. Although it looks to me like `last_commit_offset` would suffice to compute the percentage.